File: app/src/main/java/ru/taximer/taxiandroid/TaximerApp.java
package ru.taximer.taxiandroid;

import android.app.Application;
import android.content.Context;
import ru.taximer.taxiandroid.di.AppComponent;
import ru.taximer.taxiandroid.di.AppModule;
import ru.taximer.taxiandroid.di.DaggerAppComponent;
import ru.taximer.taxiandroid.di.NetworkModule;


public class TaximerApp extends Application {
    private static TaximerApp _INSTANCE = null;
    private AppComponent appComponent;

    public static TaximerApp getInstance() {
        return _INSTANCE;
    }

    @Override
    public void onCreate() {
        _INSTANCE = this;
        super.onCreate();
        appComponent = createAppComponent();
//        LeakCanary.install(this);
    }

    public static Context getContext() {
        return _INSTANCE.getApplicationContext();
    }

    private AppComponent createAppComponent() {
        return DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .networkModule(new NetworkModule())
                .build();
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }
}

File: app/src/main/java/ru/taximer/taxiandroid/TaximerApp.kt
package ru.taximer.taxiandroid


import android.content.Context
import android.support.multidex.MultiDex
import android.support.multidex.MultiDexApplication
import com.crashlytics.android.Crashlytics
import com.crashlytics.android.answers.Answers
import com.orhanobut.hawk.Hawk
import io.fabric.sdk.android.Fabric
import ru.taximer.taxiandroid.di.AppModule
import ru.taximer.taxiandroid.di.DaggerTaxiAppComponent
import ru.taximer.taxiandroid.di.TaxiAppComponent
import kotlin.system.measureTimeMillis


class TaximerApp : MultiDexApplication() {

    companion object {
        private lateinit var _appModule: AppModule
        val appModule: AppModule
            get() = _appModule

        private val _appComponent: TaxiAppComponent by lazy {
            DaggerTaxiAppComponent.builder()
                    .appModule(appModule)
                    .build()
        }

        @JvmStatic
        val appComponent: TaxiAppComponent
            get() = _appComponent
    }

    override fun onCreate() {
        super.onCreate()
        initDagger()
        Hawk.init(this).build()
        Fabric.with(this, Answers(), Crashlytics())
    }

    override fun attachBaseContext(context: Context) {
        super.attachBaseContext(context)
        MultiDex.install(this)
    }

    private fun initDagger() {
        measureTimeMillis {
            _appModule = AppModule(this)
        }
    }
}
